packages/plugin-lighthouse/src/lib/runner/constants.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  type CliFlags,
+  type Config,
+  type IcuMessage,
+  Audit as LHAudit,
+  defaultConfig,
+} from 'lighthouse';
+import { join } from 'node:path';
+import { Audit, DEFAULT_PERSIST_OUTPUT_DIR, Group } from '@code-pushup/models';
+import { LIGHTHOUSE_PLUGIN_SLUG } from '../constants';
+
+export const LIGHTHOUSE_OUTPUT_PATH = join(
+  DEFAULT_PERSIST_OUTPUT_DIR,
+  LIGHTHOUSE_PLUGIN_SLUG,
+);
+
+const { audits, categories } = defaultConfig;
+
+// internal intermediate variable to derive the relevant audits
+const allRawLighthouseAudits = await Promise.all(
+  (audits ?? []).map(async value => await loadLighthouseAudit(value)),
+);
+
+export const LIGHTHOUSE_NAVIGATION_AUDITS: Audit[] = allRawLighthouseAudits
+  // This plugin only supports the "navigation" mode of Lighthouse in the current implementation
+  // If we don't exclude other audits we throw in the plugin output validation as some of the provided audits are not included in `lighthouse-report.json`
+  .filter(
+    audit =>
+      audit.meta.supportedModes == null ||
+      (Array.isArray(audit.meta.supportedModes) &&
+        audit.meta.supportedModes.includes('navigation')),
+  )
+  .map(audit => ({
+    slug: audit.meta.id,
+    title: getMetaString(audit.meta.title),
+    description: getMetaString(audit.meta.description),
+  }));
+
+const navigationAuditSlugs = new Set(
+  LIGHTHOUSE_NAVIGATION_AUDITS.map(({ slug }) => slug),
+);
+export const LIGHTHOUSE_GROUPS: Group[] = Object.entries(categories ?? {}).map(
+  ([id, category]) => ({
+    slug: id,
+    title: getMetaString(category.title),
+    ...(category.description && {
+      description: getMetaString(category.description),
+    }),
+    refs: category.auditRefs
+      .filter(({ id: auditSlug }) => navigationAuditSlugs.has(auditSlug))
+      .map(ref => ({
+        slug: ref.id,
+        weight: ref.weight,
+      })),
+  }),
+);
+export type LighthouseGroupSlugs =
+  | 'performance'
+  | 'accessibility'
+  | 'best-practices'
+  | 'seo'
+  | 'pwa';
+
+function getMetaString(value: string | IcuMessage): string {
+  if (typeof value === 'string') {
+    return value;
+  }
+  return value.formattedDefault;
+}
+
+async function loadLighthouseAudit(
```

Comment:
👩‍💻Maintainability: Ideally, there should only be constants in `constants.ts` file.

Reply:
As this is really only related to the constants and is needed to get the constants I would keep it there.